lnwallet/channel.go

```diff
@@ -8616,7 +8616,7 @@ func CreateCooperativeCloseTx(fundingTxIn wire.TxIn,
 			Value:    int64(ourBalance),
 		})
 	}
-	if theirBalance >= remoteDust {
+	if theirBalance >= remoteDust && theirBalance >= localDust {
```

Comment:
technically, the wording of the spec means this should be `if theirBalance >= localDust`. I think some clarification in the spec meeting would be helpful here. You're right an incompatibility exists even with this change which is why I think it's important to nail down the specifics and see if we can get rid of that incompatibility for coop close v1. I don't think a partial fix is worth doing. When coop close v2 is deployed (slated for 0.18.1 but may get pushed to 0.19), this logic is likely to get superseded but may be toggle-able at first.